### PR TITLE
Intolerance Parameter

### PR DESCRIPTION
**Pull request by NSMcEnroe**
Features: Added the intolerance parameter

Possible Refactor: -There is an optional argument in the recipe facade.  This is probably a cleaner way to do this.

That's about it.  There's some if/else logic, but the API lets us keep this to a minimum.  

There are a few more methods in the facade and services, and all accompanying tests have been written. 

No merge conflicts to report.

### CODE REVIEW

File: app/controllers/api/v1/search_controller.rb [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
+    selected_recipes = specific_search
     #Possible refactor: I'm passing in the id in a weird way.
     render json: SearchedRecipeSerializer.new(selected_recipes)
-    #We'll want to enter a begin and rescue clause later
+  end
+
+  private
+
+  def specific_search
+    if params[:q] && params[:intolerances]
```
Comment (tblackmon-tiel) [inline app/controllers/api/v1/search_controller.rb:11]: Just a small comment that I've had weird interactions in conditionals like this, if that occurs check out `.present?`

**Review by tblackmon-tiel — COMMENTED** [review]
Looks solid. There's definitely a cleaner way to build the facade without two separate methods, but we're at the "just make it work" point right now, so I'm cool with it, and it's really not bad at all to begin with.

File: app/controllers/api/v1/search_controller.rb (hunk 1 above)
Comment (WillWeston94) [reply app/controllers/api/v1/search_controller.rb:11]: Also we may be able to bring this topic up in refactoring as creating some type of before_action index and then we can just set the variable in that same before action method and call it just as you have it as render json! 

Thats hopeful work though tbh

**Review by WillWeston94 — APPROVED** [review]
Nice use of context in the tests also see single line comment for hopeful refactor thought later on! Nice work!

**Pull request merged**